cns/logger/cnslogger.go

```diff
@@ -17,6 +19,9 @@ type CNSLogger struct {
 	DisableMetricLogging bool
 	DisableEventLogging  bool
 
+	ETWLogger        *zap.Logger
```

Comment:
This feels like it would benefit from being a [Chain of Responsibility](https://refactoring.guru/design-patterns/chain-of-responsibility`). It doesn't do anything dramatically different than the `logger` field in this struct (I think the method set is the same). So if that logger was like a stdout logger and this one was an ETW logger, it shouldn't matter exactly _what_ they do. There should just be a chain of loggers that get dispatched into sequentially with each log message to be logged.

Reply:
Okay, makes sense. I'm not sure why I didn't see that as a `*zap.Logger`, which would serve the same purpose.